core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSignerTest.java

```diff
@@ -343,4 +344,337 @@ public void sign_WithPayloadSigningNullAndHttp_FallsBackToPayloadSigning() {
         assertThat(signedRequest.request().firstMatchingHeader("x-amz-content-sha256"))
             .hasValue("a15c8292b1d12abbbbe4148605f7872fbdf645618fee5ab0e8072a7b34f155e2");
     }
+
+    @Test
+    public void signAsync_WithNoAdditonalProperties_DelegatesToHeaderSigner() {
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> {
+            },
+            signRequest -> {
+            }
+        );
+
+        AsyncSignedRequest signedRequest = signer.signAsync(request).join();
+
+        assertThat(signedRequest.request().firstMatchingHeader("Authorization")).isPresent();
+    }
+
+    @Test
+    public void signAsync_WithQueryAuthLocation_DelegatesToQuerySigner() {
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> {
+            },
+            signRequest -> signRequest.putProperty(AUTH_LOCATION, AuthLocation.QUERY_STRING)
+        );
+
+        AsyncSignedRequest signedRequest = signer.signAsync(request).join();
+
+        assertThat(signedRequest.request().firstMatchingRawQueryParameter("X-Amz-Signature")).isPresent();
+    }
+
+    @Test
+    public void signAsync_WithHeaderAuthLocationAndExpirationDuration_Throws() {
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> {
+            },
+            signRequest -> signRequest
+                .putProperty(AUTH_LOCATION, AuthLocation.HEADER)
+                .putProperty(EXPIRATION_DURATION, Duration.ZERO)
```

Comment:
not directly related to this change, should we validate that EXPIRATION_DURATION is non-zero? I see you are using ZERO in even a successful case earlier like sign_WithQueryAuthLocationAndExpiration_DelegatesToPresignedSigner. Maybe we can fix that separately, but for the purpose of this test, use a non zero value.

Reply:
Yeah, probably. Added validation for duration > 0